tests/test_cli.py

```diff
@@ -161,3 +163,49 @@ def test_all_fields_toml_out(mocker, capsys, make_dist):
             checksum=checksum
         )
     )
+
+
+def test_fmt_ignore_errors_on_stdout(mocker, capsys, make_dist):
+    make_dist(name="distA", install_requires=["distB1>=0.1", "distB2>=0.1"], version="0.1")
+    make_dist(name="distB1", install_requires=["distC1>=0.1", "distC2>=0.1"], version="0.1")
+    make_dist(name="distC2", version="0.1")
+    make_dist(name="distB2", version="0.1")
+
+    mocker.patch(
+        "sys.argv", "johnnydep distA --ignore-errors --fields name".split()
+    )
+    main()
+    out, err = capsys.readouterr()
+    assert err == ""
+    assert out == dedent(
+        """\
+        name
+        ----------------------------
+        distA
+        ├── distB1>=0.1
+        │   ├── distC1>=0.1 (FAILED)
+        │   └── distC2>=0.1
+        └── distB2>=0.1
+    """)
+
+
+@pytest.mark.noautofixt
+def test_ignore_errors_build_error(mocker, capsys, fake_pip):
+    import os
+    os.environ["JDT3_FAIL"] = "1"
```

Comment:
Use [`monkeypatch.setenv`](https://docs.pytest.org/en/6.2.x/monkeypatch.html#monkeypatching-environment-variables) so that this is scoped only to the necessary test.